### PR TITLE
Hw21

### PR DESCRIPTION
**Pull request by Engelko**

### CODE REVIEW

File: Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#Ubuntu
+- name: "Install Docker on Ubuntu"
+  block:
+    - name: "Set up the repository"
+      ansible.builtin.package:
+       name: "{{item}}"
+       update_cache: yes
+      loop:
+       - apt-transport-https=2.0.5
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml]: Package name and version should be two different variables

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# Copy  file .sh to hosts
+- name: "Template sh to hosts"
+  ansible.builtin.template:
+    src: /home/engelko/DevOps_course/Bash_scripts/HW21/in_hw20/roles/hw202/files/sysdlogging.sh
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: correct path

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    dest: /tmp/logdeamon.sh
+    owner: root
+    mode: '0744'
+  become: true
+
+  # Copy  file .service to hosts
+- name: "Template service.file to hosts"
+  ansible.builtin.template:
+    src: /home/engelko/DevOps_course/Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.service.j2
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: correct path

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    dest: /lib/systemd/system/logsysd.service
+    owner: root
+    mode: '0644'
+  become: true
+
+  # Copy  file .timer to hosts
+- name: "Template timer.file to hosts"
+  ansible.builtin.template:
+    src: /home/engelko/DevOps_course/Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.timer.j2
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: correct path

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml (hunk 2 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: changed

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml (hunk 3 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: changed

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml (hunk 4 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: changed

File: Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml (hunk 1 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml]: done
https://i.imgur.com/qCZbh7R.png

Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml]: https://i.imgur.com/kRalUiZ.png

File: Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.service.j2 [hunk 5]
```diff
@@ -0,0 +1,10 @@
+[Unit]
+Description=System Monitoring
+After=local-fs.target systemd-sysctl.service systemd-modules-load.service
+
+[Service]
+Type=forking
+ExecStart=/tmp/logdeamon.sh
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.service.j2]: Set as variable

File: Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.timer.j2 [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[Unit]
+Description=timer for system_monitor
+
+[Timer]
+OnUnitActiveSec=30s
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.timer.j2]: Set as variable

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+- name: "Template sh to hosts"
+  ansible.builtin.template:
+    src: sysdlogging.sh
+    dest: /tmp/logdeamon.sh
+    owner: root
+    mode: '0744'
+  become: true
+
+  # Copy  file .service to hosts
+- name: "Template service.file to hosts"
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: use only one task for both templates

Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: fix

File: Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.timer.j2 (hunk 6 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.timer.j2]: fix

File: Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.service.j2 (hunk 5 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/templates/logsysd.service.j2]: fix

File: Bash_scripts/HW21/in_hw20/group_vars/centos/centos_var.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+pack_git_pyt:
+ - { name: "git-", version: "1.8.3.1-23.el7_8" }
+ - { name: "python3-", version: "3.6.8-18.el7" }
+pack_docker:
+ - { name: "docker-ce-", version: "20.10.6-3.el7" }
+ - { name: "docker-ce-cli-", version: "20.10.6-3.el7" }
+ - { name: "containerd.io-", version: "1.4.4-3.1.el7" }
+url_key:
+ - { url: "https://download.docker.com/linux/centos/docker-ce.repo", dest: "/etc/yum.repos.d/docer-ce.repo" }
+
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/group_vars/centos/centos_var.yml]: new line!

File: Bash_scripts/HW21/in_hw20/hosts.yml [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#Server list
+
+all:
+ children:
+  linux:
+   hosts:
+    192.168.1.31:
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/hosts.yml]: use groups instead of IPs

File: Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+  when: ansible_facts['os_family'] == "RedHat"
+
+- name: "Install Docker Engine"
+  ansible.builtin.package:
+   name: "{{ item.name }}{{ item.version }}"
+   update_cache: yes
+  with_items:
+    - "{{ pack_docker }}"
+  notify: "Docker start"
+
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml]: line

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+# Copy  files to hosts for systemd logs
+- name: "Template files to hosts"
+  ansible.builtin.template:
+    src: "{{ item.src }}"
+    dest: "{{ item.dest }}"
+    owner: "{{ item.owner }}"
+    mode: "{{ item.mode }}"
+  with_items:
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: replace with loop as ansible recommended

File: Bash_scripts/HW21/in_hw20/group_vars/centos/centos_var.yml (hunk 8 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/group_vars/centos/centos_var.yml]: fix

File: Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml (hunk 10 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw201/tasks/docker.yml]: fix

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml (hunk 11 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/roles/hw202/tasks/templatesysd.yml]: done

File: Bash_scripts/HW21/in_hw20/hosts.yml (hunk 9 above)
Comment (Engelko) [reply Bash_scripts/HW21/in_hw20/hosts.yml]: done

File: Bash_scripts/HW21/in_hw20/roles/hw202/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+   - "{{ path_files }}"
+
+#Copy  files to hosts for systemd logs
+- name: "Push files to hosts"
+  import_tasks: templatesysd.yml
+  become: true
+
+#Start service systemd logging
+- name: "Start logsysd"
+  systemd:
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw202/tasks/main.yml:26]: Use the same stile. If you are using full path to modules use it everywhere.

File: Bash_scripts/HW21/in_hw20/roles/hw201/tasks/main.yml [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  import_tasks: install.yml
+  become: true
+  
+#Install docker
+- name: Install docker
+  import_tasks: docker.yml
+  become: true 
+
+#  Optional sudo or wheel group for user hw20
+# If the value "{{ group_sudo }}" is not specified, the group will not be added
```
Comment (mavrik90) [inline Bash_scripts/HW21/in_hw20/roles/hw201/tasks/main.yml:12]: Nice one!

**Review by mavrik90 — APPROVED** [review]
Cool! Good for you!

**Pull request merged**